File: 1-two-sum/two-sum.go
func twoSum(nums []int, target int) []int {
    mp := make(map[int]int)
    for i,val := range(nums){
        if val,ok := mp[target - val]; ok {
            return []int{val, i}
        }
        mp[val] = i;
    }
    
    return []int{}
}

File: 1-two-sum/two-sum.kt
class Solution {
    fun twoSum(nums: IntArray, target: Int): IntArray {
        val mp = mutableMapOf<Int,Int>()
        for(i in 0..nums.size-1){
            if(mp.containsKey(target-nums[i])){
                return intArrayOf(mp[target-nums[i]]!!,i)
            }
            mp.put(nums[i],i)
        }
        return intArrayOf()
    }
}